### PR TITLE
api: use hex encoding in postage api

### PR DESCRIPTION
**Pull request by ralph-pichler**
api returned base64 encoded id instead of hex.

### CODE REVIEW

File: pkg/api/postage.go [hunk 1]
```diff
@@ -10,11 +10,12 @@ import (
 	"strconv"
 
 	"github.com/ethersphere/bee/pkg/jsonhttp"
+	"github.com/ethersphere/bee/pkg/swarm"
 	"github.com/gorilla/mux"
 )
 
 type postageCreateResponse struct {
-	BatchID []byte `json:"batchID"`
+	BatchID swarm.Address `json:"batchID"`
```
Comment (acud) [inline pkg/api/postage.go:18]: not sure we'd actually want to use this type here, strictly speaking, this isn't an `address`

Comment (ralph-pichler) [reply pkg/api/postage.go:18]: i only used this because it has the right encoding. `[]byte` uses base64 unfortunately.

Comment (acud) [reply pkg/api/postage.go:18]: `[]byte` and implement `json.Marshaler`?

**Review by acud — APPROVED** [review]

File: pkg/api/postage.go (hunk 1 above)
Comment (ralph-pichler) [reply pkg/api/postage.go]: you cannot implement anything on `[]byte`. I made a custom type for this field.

Comment (acud) [reply pkg/api/postage.go]: i meant the define the JSON marshaler on `postageCreateResponse`

**Review by acud — APPROVED** [review]

**Pull request merged**